src/py/flwr/common/message.py

```diff
@@ -309,6 +304,26 @@ def create_error_reply(self, error: Error, ttl: float | None = None) -> Message:
             )
             message.metadata.ttl = ttl
 
+        # Limit the reply TTL to not exceed the
+        # expiration time of the message it replies to.
+        # Condition: msg.metadata.ttl + msg.metadata.created_at ≥
+        #            reply_msg.metadata.ttl + reply_msg.metadata.created_at
+        # A small tolerance is introduced to account
+        # for floating-point precision issues.
+        max_allowed_ttl = (
+            self.metadata.created_at + self.metadata.ttl - message.metadata.created_at
+        )
+        if message.metadata.ttl - max_allowed_ttl > TTL_TOLERANCE:
+            log(
+                WARNING,
+                "The reply TTL of %.2f seconds exceeded the "
+                "allowed maximum of %.2f seconds."
+                "The TTL has been updated to the allowed maximum.",
+                ttl,
+                max_allowed_ttl,
+            )
+            message.metadata.ttl = max_allowed_ttl
```

Comment:
This block of code and the one below should live in a shared function to reduce code duplication.